org.mixedrealitytoolkit.input/Interactors/Gaze/TrackedPoseDriverWithFallback.cs

```diff
@@ -0,0 +1,98 @@
+        /// </summary>
+        public InputActionProperty FallbackRotationAction => fallbackRotationAction;
+
+        [SerializeField, Tooltip("The fallback Input System action to get the Tracking State for this GameObject when the default track status action has no data. If not specified, this will fallback to the device's tracking state that drives the position or rotation action. Must be a IntegerControl Control.")]
+        private InputActionProperty fallbackTrackingStateAction;
+
+        /// <summary>
+        /// The fallback Input System action to get the Tracking State for this GameObject when the default track status
+        /// action has no data. If not specified, this will fallback to the device's tracking state that drives the position
+        /// or rotation action. Must be a IntegerControl Control.
+        /// </summary>
+        public InputActionProperty FallbackTrackingStateAction => fallbackTrackingStateAction;
+
+        #endregion Fallback action values
+
+        #region TrackedPoseDriver Overrides 
+        /// <inheritdoc />
+        protected override void PerformUpdate()
+        {
+            base.PerformUpdate();
+
+            if (trackingStateInput.action == null)
+            {
+                Debug.LogWarning("TrackedPoseDriverWithFallback.trackingStateInput.action is null, no fallback will be used.");
+                return;
+            }
+
+            var positionAction = fallbackPositionAction.action;
+            var hasPositionAction = positionAction != null;
+            var hasPositionFallbackAction = fallbackPositionAction != null;
+
+            var rotationAction = fallbackRotationAction.action;
+            var hasRotationAction = rotationAction != null;
+            var hasRotationFallbackAction = fallbackRotationAction != null;
+
+            InputTrackingState inputTrackingState = (InputTrackingState)trackingStateInput.action.ReadValue<int>();
+
+            if (!inputTrackingState.HasFlag(InputTrackingState.Position) && !inputTrackingState.HasFlag(InputTrackingState.Rotation) && FallbackTrackingStateAction.action != null)
+            {
+                inputTrackingState = (InputTrackingState)FallbackTrackingStateAction.action.ReadValue<int>();
+            }
+
+            // If no position data then swap the position action with the fallback position action if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Position) && hasPositionAction && hasPositionFallbackAction)
+            {
+                (fallbackPositionAction, positionInput) = (positionInput, fallbackPositionAction);
+            }
+
+            // If no rotation data then swap the rotation action with the fallback rotation action if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Rotation) && hasRotationAction && hasRotationFallbackAction)
+            {
+                (fallbackRotationAction, rotationInput) = (rotationInput, fallbackRotationAction);
+            }
```

Comment:
I see what you're trying to do here...prevent failures in next update loop.  However, we shouldn't be changing public property values from under the consumers of this class.  Consumers can be caught off guard if these values change.

Also, with this there will be bugs.  If eye tracking is lost briefly, the app will be stuck in head tracking mode.  The app should revert back to eye tracking once available.   

Please revert this swapping logic

Reply:
I see your point.  I'm not sure I can just go back to the pre-XRI3 logic because we use the XRControllerState to replace the missing data with the fallback data and we are trying to stop using XRControllerState.  As in:
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/d36d873e-8693-480b-b962-271dca132f96)
On the other hand, I did try intially to replicate the original logic but couldn't find a way to update the replace missing values directly in InputTrackingState nor the Actions themselves, for example:
![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/5c8b5585-2400-49dd-abb0-d446e69cc1e6)
Do you know a way to accomplish this without the XRControllerState?

Reply:
@AMollis:  I haven't been able to find a way to re-implement the original logic without an XRControllerState so I switched approaches: use InputActions with fallbacks and let the Input System get data from a priority list of bindings.  I provide more details in the Description of this PR.  What do you think of this alternate approach?

Reply:
Thank you for synching with me on this issue.  I've implemented the approach we discussed in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/746/commits/23aec29f02753d453751f0feefe4130a3d6c8f2d and it works (steps on how I tested are in the description of this PR), resolving.